server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -208,6 +210,21 @@ public DateTime(StreamInput in) throws IOException {
                 this.timeZone = ZoneId.of(zoneId);
                 this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
             }
+            if (in.getVersion().onOrAfter(Version.CURRENT)) {
+                //if stream is from 7.7 it will have a flag indicating if format is joda
+                boolean isJoda = in.readBoolean();
+                this.formatter = isJoda ? Joda.forPattern(datePattern) : DateFormatter.forPattern(datePattern);
+            } else if (Joda.isJodaPattern(in.getVersion(), datePattern)) {
+                //when received a stream from 6.0-6.latest it can be java if starts with 8 otherwise joda
+                this.formatter = Joda.forPattern(datePattern);
+            }else{
+                // unknown if this is joda or java for versions earlier then [7.0-7.7).
+                //todo consider throwing exception.. .
```

Comment:
this will happen when there is a mixed env with 7.7 and 7.0. Overall I think it is ok to assume it is java. most of the time it would work so no point breaking too many people